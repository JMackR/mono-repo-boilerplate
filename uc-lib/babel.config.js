module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
