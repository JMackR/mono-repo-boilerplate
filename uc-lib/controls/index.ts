export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
