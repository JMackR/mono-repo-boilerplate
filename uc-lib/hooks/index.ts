export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
