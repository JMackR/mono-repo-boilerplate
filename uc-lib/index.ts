export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
