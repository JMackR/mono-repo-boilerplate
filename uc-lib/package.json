{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
