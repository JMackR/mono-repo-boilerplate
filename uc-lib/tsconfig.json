{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
