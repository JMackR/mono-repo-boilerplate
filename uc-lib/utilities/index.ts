export * from "./dimensions-utility"
export * from "./media-query"
