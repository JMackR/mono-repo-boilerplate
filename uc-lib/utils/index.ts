export * from "./iif"
export * from "./format-number"
