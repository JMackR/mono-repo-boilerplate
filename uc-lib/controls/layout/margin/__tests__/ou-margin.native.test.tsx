import React from "react"
import { render } from "react-native-testing-library"
import { Margin, Text } from "uc-lib"

describe("Margin Snapshot Tests", () => {
  test("marginStep renders correctly", () => {
    const tree = render(
      <Margin marginStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginTopStep renders correctly", () => {
    const tree = render(
      <Margin marginTopStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginLeftStep renders correctly", () => {
    const tree = render(
      <Margin marginLeftStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginBottomStep renders correctly", () => {
    const tree = render(
      <Margin marginBottomStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginRightStep renders correctly", () => {
    const tree = render(
      <Margin marginRightStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginTopStep renders correctly over marginStep", () => {
    const tree = render(
      <Margin marginStep={1} marginTopStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginLeftStep renders correctly over marginStep", () => {
    const tree = render(
      <Margin marginStep={1} marginLeftStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginBottomStep renders correctly over marginStep", () => {
    const tree = render(
      <Margin marginStep={1} marginBottomStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
  test("marginRightStep renders correctly over marginStep", () => {
    const tree = render(
      <Margin marginStep={1} marginRightStep={2}>
        <Text>Example String</Text>
      </Margin>,
    )
    expect(tree).toMatchSnapshot()
  })
})
