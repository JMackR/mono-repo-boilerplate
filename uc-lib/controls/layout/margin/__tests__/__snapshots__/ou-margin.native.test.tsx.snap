// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Margin Snapshot Tests marginBottomStep renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": undefined,
      "marginBottom": 8,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginBottomStep renders correctly over marginStep 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": 4,
      "marginBottom": 8,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginLeftStep renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": 8,
      "marginRight": undefined,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginLeftStep renders correctly over marginStep 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": 4,
      "marginBottom": undefined,
      "marginLeft": 8,
      "marginRight": undefined,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginRightStep renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": 8,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginRightStep renders correctly over marginStep 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": 4,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": 8,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginStep renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": 8,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginTopStep renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": 8,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;

exports[`Margin Snapshot Tests marginTopStep renders correctly over marginStep 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexGrow": 0,
      "flexShrink": 1,
      "height": undefined,
      "justifyContent": "flex-start",
      "margin": 4,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": 8,
      "width": undefined,
    }
  }
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Example String
  </Text>
</View>
`;
